packages/core/core/src/AssetGraphBuilder.js

```diff
@@ -214,8 +216,18 @@ export default class AssetGraphBuilder extends EventEmitter {
   }
 
   async validate(): Promise<void> {
+    // Schedule validations on workers for all plugins that implement the one-asset-at-a-time "validate" method.
     let promises = this.assetRequests.map(request =>
-      this.runValidate({request, optionsRef: this.optionsRef}),
+      this.runValidate({requests: [request], optionsRef: this.optionsRef}),
+    );
+    // Schedule validations on the main thread for all validation plugins that implement "validateAll".
+    promises.push(
```

Comment:
Ideally we would run validateAll on a dedicated worker instead of the main thread. Although we probably need to benchmark if it's worth it. (as in it's pointless to multithread if main thread is idle)

Reply:
I agree this is a good idea, and I'm happy to work on it. @padmaia mentioned [above](https://github.com/parcel-bundler/parcel/pull/4237#issuecomment-595572660) that you may have done something similar in v1. I'm sure I can figure it out the hard way, but do you have any pointers or examples I could look at to help?

Reply:
What I did in v1 is something quite different I just created a wrapper around chokidar which started up a worker.

What I wanted to do for this is assign a worker to it, every worker has an id so in theory we should be able to send along an id to a request to the workerfarm and we should be able to only execute that on that exact worker.

But it would require quite some refactoring so we definitely shouldn't be doing this as part of this PR and open a seperate PR for it instead.